OpenRA.Game/MiniYaml.cs

```diff
@@ -141,6 +141,34 @@ public MiniYaml WithNodesAppended(IEnumerable<MiniYamlNode> nodes)
 			return new MiniYaml(Value, newNodes);
 		}
 
+		public MiniYamlNode NodeWithKey(string key)
+		{
+			var result = NodeWithKeyOrDefault(key);
+			if (result == null)
+				throw new InvalidDataException($"No node with key '{key}'");
+			return result;
+		}
+
+		public MiniYamlNode NodeWithKeyOrDefault(string key)
+		{
+			// PERF: Avoid LINQ.
+			var first = true;
+			MiniYamlNode result = null;
+			foreach (var node in Nodes)
+			{
+				if (node.Key != key)
+					continue;
+
+				if (!first)
+					throw new InvalidDataException($"Duplicate key '{node.Key}' in {node.Location}");
+
+				first = false;
+				result = node;
```

Comment:
Why not just break once the first was found? Couldn't the reader better check for duplicate keys. Or insert.

Reply:
Having duplicates checked in the helper helps keep things consistent. For example the two TS files that required fixing could have been detected at the caller, but in practise nobody remembers to do such checks. If the helper does it then it's done by everybody.